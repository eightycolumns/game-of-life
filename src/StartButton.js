import React from "react";

class StartButton extends React.Component {
  render() {
    return (
      <div className="StartButton">
        <button onClick={this.props.onClick}>Start</button>
      </div>
    );
  }
}

export default StartButton;
