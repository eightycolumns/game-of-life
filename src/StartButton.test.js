import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import React from "react";

import StartButton from "./StartButton";

Enzyme.configure({
  adapter: new Adapter()
});

describe("a StartButton component", () => {
  it("renders correctly", () => {
    const startButton = shallow(
      <StartButton onClick={function startGame() {}} />
    );

    expect(startButton).toMatchSnapshot();
  });
});
