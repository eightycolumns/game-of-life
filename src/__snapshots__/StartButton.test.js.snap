// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a StartButton component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartButton
    onClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[Function]}
>
        Start
</button>,
      "className": "StartButton",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Start",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Start",
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[Function]}
>
          Start
</button>,
        "className": "StartButton",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Start",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Start",
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
